File: Vika1/Python/guess.py
import sys

mid = 500
min = 1
max = 1001

for x in range(10):
    print(mid)
    sys.stdout.flush()
    resp = input()
    if(resp == "correct"): break
    if(resp == "higher"): min = mid
    if(resp == "lower"): max = mid
    mid = min + (max - min)//2

File: Vika1/c++/guess.cpp
#include <iostream>
#include <string>


int main(){
    int mid = 500;
    int low = 1;
    int max = 1001;
    std :: string resp;

    for(int i = 0; i < 10; i++){
        std :: cout << mid << "\n";
        std :: cin >> resp;

        if(resp[0] == 'c') break;
        if(resp[0] == 'l') max = mid;
        if(resp[0] == 'h') low = mid;

        mid = low + (max - low)/2;

    }
}